test/integration/filters/test_listener_filter.cc

```diff
@@ -0,0 +1,34 @@
+#include "test/integration/filters/test_listener_filter.h"
+
+namespace Envoy {
+
+/**
+ * Config registration for the test filter.
+ */
+class TestInspectorConfigFactory : public Server::Configuration::NamedListenerFilterConfigFactory {
+public:
+  // NamedListenerFilterConfigFactory
+  Network::ListenerFilterFactoryCb createListenerFilterFactoryFromProto(
+      const Protobuf::Message& /*message*/,
+      const Network::ListenerFilterMatcherSharedPtr& listener_filter_matcher,
+      Server::Configuration::ListenerFactoryContext& /*context*/) override {
+    return [listener_filter_matcher](Network::ListenerFilterManager& filter_manager) -> void {
+      filter_manager.addAcceptFilter(listener_filter_matcher,
+                                     std::make_unique<TestListenerFilter>());
+    };
+  }
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return ProtobufTypes::MessagePtr{new Envoy::ProtobufWkt::Struct()};
```

Comment:
Given #20397 and upcoming #20049 , can we add a real protobuf for this test filter?

Reply:
I mean it's not needed here, so would it make sense to add it as a follow up over in one of those PRs?

Reply:
oh whoops, I was looking at the second and not the first.  :-/

Reply:
Yeah, we have https://github.com/envoyproxy/envoy/issues/20476 for general fix-ups.